Source/JavaScriptCore/b3/air/AirCode.cpp

```diff
@@ -46,8 +46,8 @@ static void defaultPrologueGenerator(CCallHelpers& jit, Code& code)
 {
     jit.emitFunctionPrologue();
     if (code.frameSize()) {
-        AllowMacroScratchRegisterUsageIf allowScratch(jit, isARM64());
-        jit.addPtr(MacroAssembler::TrustedImm32(-code.frameSize()), MacroAssembler::framePointerRegister,  MacroAssembler::stackPointerRegister);
+        AllowMacroScratchRegisterUsageIf allowScratch(jit, isARM64() || isARM());
+        jit.subPtr(MacroAssembler::TrustedImm32(code.frameSize()), MacroAssembler::stackPointerRegister);
```

Comment:
Why is this needed?

Reply:
This is to get a shorter encoding--there's a one-word encoding for `sub <imm>, sp -> sp` for thumb; the negative immediate and the three-operand form need more instructions to encode

... and as far as I can tell, shouldn't negatively affect the encoded length on aarch64 or x86_64 (where, if i am reading right, the change saves a redundant `mov rsp, rbp`)

Reply:
OK, so fp = sp because jit.emitFunctionPrologue does that, so this should be equivalent. Sweet.